raiden/transfer/channel.py

```diff
@@ -1369,11 +1525,12 @@ def create_sendexpiredlock(
     channel_identifier: ChannelID,
     recipient: Address,
 ) -> Tuple[Optional[SendLockExpired], Optional[MerkleTreeState]]:
-    nonce = get_next_nonce(sender_end_state)
     locked_amount = get_amount_locked(sender_end_state)
     balance_proof = sender_end_state.balance_proof
     updated_locked_amount = TokenAmount(locked_amount - locked_lock.amount)
 
+    nonce = get_next_nonce(sender_end_state)
```

Comment:
I suspect we should also store the new `nonce` in `sender_end_state` here?

Reply:
unless that happens in 1589 `channel_state.our_state.nonce = send_lock_expired.balance_proof.nonce` ?

shouldn't we keep the same pattern as `create_sendlockedtransfer(...)`, where we get the new `nonce` and update the endstate?

Reply:
I have changed this so that the caller sets the nonce for both `create_sendlockedtransfer` and `create_sendexpiredlock`. The reason i am opting for this is:
1. Giving the caller the choice of updating the state nonce if the message is to be certainly sent. Or otherwise, it would just discard the result of `create_*` call.
2. Some tests call these functions on behalf of the partner to simulate receiving those messages from the partner's node. This increments their state nonce which leads to an invalid message / state change being received.